nautobot/extras/filters/mixins.py

```diff
@@ -219,11 +219,16 @@ def _append_relationships(self, model):
         """
         Append form fields for all Relationships assigned to this model.
         """
-        source_relationships = Relationship.objects.filter(source_type=self.obj_type, source_hidden=False)
-        self._append_relationships_side(source_relationships, RelationshipSideChoices.SIDE_SOURCE, model)
-
-        dest_relationships = Relationship.objects.filter(destination_type=self.obj_type, destination_hidden=False)
-        self._append_relationships_side(dest_relationships, RelationshipSideChoices.SIDE_DESTINATION, model)
+        query = Q(source_type=self.obj_type, source_hidden=False) | Q(
+            destination_type=self.obj_type, destination_hidden=False
+        )
+        relationships = Relationship.objects.select_related("source_type", "destination_type").filter(query)
+
+        for rel in relationships.iterator():
+            if rel.source_type == self.obj_type:
+                self._append_relationships_side([rel], RelationshipSideChoices.SIDE_SOURCE, model)
+            if rel.destination_type == self.obj_type:
```

Comment:
Maybe?

```suggestion
            if rel.source_type == self.obj_type and not rel.source_hidden:
                self._append_relationships_side([rel], RelationshipSideChoices.SIDE_SOURCE, model)
            if rel.destination_type == self.obj_type and not rel.destination_hidden:
```

Reply:
That is already covered in the query, no? Anything with `{side}_hidden` should not be in the queryset. See:

```python
relationships = Relationship.objects.select_related("source_type", "destination_type").filter(query)
```

Reply:
Gotcha. Thanks for clarifying.